### DISCUSSION TITLE
How to switch gears in manual mode

### DISCUSSION THREAD

**Discussion by kutu** [jrouwe/JoltPhysics.js#59] [Q&A] [answered]
setting manual mode
```js
controllerSettings.mTransmission.mMode = Jolt.ETransmissionMode_Manual
```
later when pressing a button, i'm holding state that gear needs to be shifted
and in update call i do
```js
if (shiftUp) {
  transmission = controller.GetTransmission();
  transmission.Set(transmission.GetCurrentGear() + 1, 1);
  console.log(transmission.GetCurrentGear()); // returns 1
  console.log(controller.GetTransmission().GetCurrentGear()); // still returns 0
}
if (right || forward || brake || handBrake || shiftUp) {
  bodyInterface.ActivateBody(carBody.GetID());
}
```
maybe problem is here
![image](https://github.com/jrouwe/JoltPhysics.js/assets/228802/d386cad8-84f2-4017-b50f-0e37d6c46095)
maybe js version returns _not writable_ interface

`idl` says `Value`
```
        [Value] VehicleTransmission GetTransmission();
```
https://github.com/jrouwe/JoltPhysics.js/blob/b207e232d6ec869c8d38912f9b6ba449318b8d3c/JoltJS.idl#L2446C2-L2446C48

**Comment by jrouwe** [comment, marked as answer]
Yes, you're completely right. I just spotted this myself too and fixed it in https://github.com/jrouwe/JoltPhysics.js/commit/0f2a19f892f4419d3235c0eccb96769291640968. I'll release a new version in a moment.

**Comment by jrouwe** [comment]
New version has been released and I've verified that your code example above works now.

**Comment by jrouwe** [comment]
Thanks for reporting these!
